### PR TITLE
Fix PPFormulaNet training-loss double-shift in encoder-decoder forward

### LINKED ISSUE #47317: PPFormulaNet training-loss bug
### System Info

PPFormulaNet training-loss bug — ForCausalLMLoss incorrectly shifts labels in encoder-decoder model
The problem: PPFormulaNetForConditionalGeneration.forward() uses self.loss_function() (line 491 in modular_pp_formulanet.py) which maps to ForCausalLMLoss. ForCausalLMLoss internally shifts labels left by 1 (labels[..., 1:]), but PPFormulaNet is an encoder-decoder model where the logits are already aligned with labels — no shift is needed.
This means the model trains against misaligned labels, losing the last target token.
Where:
# modular_pp_formulanet.py, line 491
loss = self.loss_function(
    logits=logits, labels=labels, vocab_size=self.config.text_config.vocab_size, **kwargs
)

The fix (same pattern as Florence2 after #46898):
loss = self.loss_function(
    logits=logits, labels=labels, vocab_size=self.config.text_config.vocab_size,
    shift_labels=labels, **kwargs
)

Passing shift_labels=labels tells ForCausalLMLoss to skip its internal shift, keeping labels aligned with logits.
Context: Florence2 (which PPFormulaNet inherits from) was already fixed in PR #46898 with this same pattern, but PPFormulaNet overrides forward() so the fix didn't propagate. eliaghazal also flagged PPFormulaNet as potentially affected in PR #47090 discussion, but no one has filed or fixed it yet.

### Who can help?

_No response_

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

.

### Expected behavior

.

### PR DESCRIPTION
**Pull request by ShamSaleem**
# What does this PR do?

Fixes #47317

`PPFormulaNetForConditionalGeneration.forward()` computes its loss with
`self.loss_function()` → `ForCausalLMLoss`, which shifts labels left by one
(`labels[..., 1:]`) when `shift_labels` is not supplied. PPFormulaNet is an
**encoder-decoder** model, so its logits are already aligned with `labels` and
no shift is needed. The extra shift drops the last target token and trains the
model against misaligned labels.

This passes `shift_labels=labels` to `ForCausalLMLoss`, which takes the
"already-shifted" path and leaves the labels aligned with the logits.

This is the **same fix already merged for Florence2 in #46898** (and Moonshine
before it). PPFormulaNet's modular model inherits from
`Florence2ForConditionalGeneration` but **overrides `forward()`**, so the
Florence2 fix never propagated to it — `eliaghazal` flagged PPFormulaNet as
likely-affected in the #47090 discussion.

Change is applied to `modular_pp_formulanet.py` (source of truth) and the
regenerated `modeling_pp_formulanet.py`.

## Before submitting
- [ ] This PR fixes a typo or improves the docs.
- [x] Did you read the [contributor guideline](https://huggingface.co/docs/transformers/contributing) and the [PR checks](https://huggingface.co/docs/transformers/pr_checks)?
- [x] Was this discussed/approved via a Github issue? Yes — #47317.
- [ ] Did you make sure to update the documentation?
- [ ] Did you write any new necessary tests? — No new test, matching the merged precedent #46898 (source-only); happy to add a loss-alignment regression test if preferred.

> Note: AI assistance was used to help locate and draft this fix.

## Who can review?

@zucchini-nlp (reviewed & merged the identical Florence2 fix #46898)

### CODE REVIEW

**Comment by zucchini-nlp** [comment]
Already being fixed on another PR, closing as duplicate
